### DISCUSSION TITLE
controlling notebook execution order

### DISCUSSION THREAD

**Discussion by rabernat** [executablebooks/meta#743] [Q&A]
We have artifacts generated by one notebook that are needed by other dependent notebooks (see https://github.com/m2lines/L96_demo/issues/47 for details). We would like to be able to control the order of execution of our notebooks. For example, just executing them in the order they appear in the TOC would be fine. Currently the order appears to be random.

Is it possible to control execution order somehow?

**Comment by choldgraf** [comment]
Hmm, I believe that the parsing and execution happens inside of the MySTNB "MyST Parser":

https://github.com/executablebooks/MyST-NB/blob/5e2d59c4ab5e66b4522a7b12a2ef01ab90b93ba8/myst_nb/sphinx_.py#L138

So I'd assume that this would be called in the order that each page is parsed. That said, some of the execution engines might happen at different parts of the build process.

A few questions:

- Is the order of execution deterministic? If you delete the `_build` folder, then build the docs, then repeat the process, is the order the same?
- Does this change if you set a different execution engine like `force`?

**Reply by rabernat** [reply]
Thanks for the suggestions. We will investigate this. For reference, we are using `execute_notebooks: cache`. Does that mean that the execution order is controlled by jupyter-cache?

**Reply by choldgraf** [reply]
yep I believe that is correct
